### PR TITLE
[T-3055][16.0][ADD] product_search_category_attribute: base module for searching products for cat. and attr.

### PR DESCRIPTION
**Pull request by manuelregidor**
@ValentinVinagre @HaraldPanten 
@luis-ron **It is totally normal that the first test fails** (@manuel-florido is aware of it). Could you review, please? You should test this module along with product_search_category_attribute_sale (https://github.com/sygel-technology/sy-sale-workflow/pull/47) and product_search_category_attribute_stock (https://github.com/sygel-technology/sy-stock-logistics-workflow/pull/16), as this one does nothing on its own.

### CODE REVIEW

**Review by luis-ron — APPROVED** [review]
Functional review: LGTM 👍🏻

**Comment by manuelregidor** [comment]
@Tisho99 Could you review, please?

File: product_search_category_attribute/views/product_search_category_attribute_views.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                                    widget="many2many_tags"
+                                    force_save="1"
+                                    options="{'no_create_edit': True, 'no_create': True, 'no_open': True}"
+                                />
+                            </tree>
+                        </field>
+                    </page>
+                <footer>
+                    <button name="cancel" string="Cancel" special="cancel" />
+                </footer>
```
Comment (Tisho99) [inline product_search_category_attribute/views/product_search_category_attribute_views.xml]:
```suggestion
                    <footer>
                        <button name="cancel" string="Cancel" special="cancel" />
                    </footer>
```

**Review by Tisho99 — APPROVED** [review]
LGTM. Code review

Minor suggestions

File: product_search_category_attribute/models/product_search_category_attribute.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductSearchCatAttr(models.AbstractModel):
+    _name = "product.search.cat.attr"
+    _description = "Product Search Category Attribute"
+
+    RELOAD_VIEW = ""
```
Comment (ValentinVinagre) [inline product_search_category_attribute/models/product_search_category_attribute.py:11]: this reload_view why is a general field of the model?

File: product_search_category_attribute/models/product_search_category_attribute.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                        "product_attribute_value_id"
+                    ).ids
+                )
+            products = attr_products
+        return products
+
+    def search_products(self):
+        raise NotImplementedError()
+
+    @api.depends("applied_category_ids")
```
Comment (ValentinVinagre) [inline product_search_category_attribute/models/product_search_category_attribute.py:124]: Try to do only one write in a function and not multiple writes, for debugging, extension, etc. it is always better.

```
@api.depends("applied_category_ids")
    def _compute_applied_category(self):
        for sel in self:
            applied_category = False
            if sel.applied_category_ids:
                applied_category = sel.applied_category_ids[-1].category_id.id
           sel.applied_category_ids = applied_category_ids
```

**Review by ValentinVinagre — CHANGES_REQUESTED** [review]
Some things

File: product_search_category_attribute/models/product_search_category_attribute.py (hunk 2 above)
Comment (manuelregidor) [reply product_search_category_attribute/models/product_search_category_attribute.py:11]: Refer to def reload_view(self) method. It is a class attribute that needs to be defined in all the classes that inherit this one.

**Comment by manuelregidor** [comment]
@ValentinVinagre changes done

File: product_search_category_attribute/models/product_search_category_attribute.py (hunk 2 above)
Comment (ValentinVinagre) [reply product_search_category_attribute/models/product_search_category_attribute.py:11]: ahhh okay, I see.

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**